### PR TITLE
internal/e2e: Add Header Condition tests to E2E route

### PR DESCRIPTION
**Pull request by stevesloka**
Signed-off-by: Steve Sloka <steve@stevesloka.com>

### CODE REVIEW

File: internal/e2e/rds_test.go [hunk 1]
```diff
@@ -2579,6 +2582,17 @@ func websocketroute(c string) *envoy_api_v2_route.Route_Route {
 	return cl
 }
 
+func headerroute(c string) *envoy_api_v2_route.Route_Route {
+	cl := routecluster(c)
+	//cl.Route_mat
```
Comment (davecheney) [inline internal/e2e/rds_test.go]: nit: looks like debugging breadcrumb

Comment (stevesloka) [reply internal/e2e/rds_test.go]: Turns out the whole method isn't needed. =)

**Comment by stevesloka** [comment]
I think there's an ordering problem with the tests and how the results are compared to the desired, sometimes they work, sometimes they fail.

**Comment by davecheney** [comment]
```
                        envoy.Route(envoy.RoutePrefix("/"), routecluster("default/svc1/80/da39a3ee5e")),
                        envoy.Route(envoy.RoutePrefix("/", dag.HeaderCondition{
                                Name:      "x-header",
                                Value:     "abc",
                                MatchType: "exact",
                                Invert:    false,
                        }), routecluster("default/svc2/80/da39a3ee5e")),
```
I think the problem is the sorting of routes doesn't take headers into account. Let me see if I can fix this.

**Pull request merged**